src/jdk.attach/linux/native/libattach/VirtualMachineImpl.c

```diff
@@ -120,6 +160,8 @@ JNIEXPORT void JNICALL Java_sun_tools_attach_VirtualMachineImpl_connect
 JNIEXPORT void JNICALL Java_sun_tools_attach_VirtualMachineImpl_sendQuitTo
   (JNIEnv *env, jclass cls, jint pid)
 {
+    // Only give up sending SIGQUIT if we see that SigCgt is not set.
+    if (check_sigquit_caught(pid) == 0) return;
```

Comment:
Suggestion (assumes bool function):
```
// Only send the SIGQUIT if we can see that the target JVM is ready to catch it.
if (check_sigquit_caught(pid) && kill((pid_t)pid, SIGQUIT) != 0) {
    JNU_ThrowIOExceptionWithLastError(env, "kill");
}
```

Reply:
The reason I would leave retval -1 because I guess someone may disable procfs entirely with kernel configure. As a result, we never know that answer of `check_sigquit_caught` for sure.

Reply:
But if you never check for -1 you will never know that. The code only checks for zero or not zero and that is a boolean result.